internal/loader/configuration_setting_loader.go

```diff
@@ -230,17 +245,26 @@ func (csl *ConfigurationSettingLoader) CreateKeyValueSettings(ctx context.Contex
 				}
 				switch *setting.ContentType {
 				case FeatureFlagContentType:
-					continue // ignore feature flag at this moment, will support it in later version
-				case KeyVaultReferenceContentType:
+					continue // ignore feature flag while getting key value settings
+				case SecretReferenceContentType:
 					if setting.Value == nil {
 						return nil, fmt.Errorf("The value of Key Vault reference '%s' is null", *setting.Key)
 					}
-					if csl.Spec.Secret == nil {
+
+					var secretType corev1.SecretType = corev1.SecretTypeOpaque
+					var err error
+					if secretTypeTag, ok := setting.Tags[PreservedSecretTypeTag]; ok {
+						secretType, err = parseSecretType(secretTypeTag)
```

Comment:
`parseSecretType` looks like only supporting tls type? Can user set opaque type through tag?

Reply:
No, current I don't want to support it, if someone want an opaque secret, just don't set any tag is fine